src/tip-opengov.ts

```diff
@@ -55,29 +50,46 @@ export async function tipOpenGov(opts: { state: State; api: ApiPromise; tipReque
   });
 
   if (tipResult.success && polkassembly) {
+    let referendumId: number;
+    try {
+      const maybeReferendumId = await getReferendumId(await api.at(tipResult.blockHash), encodedProposal);
+      if (maybeReferendumId === undefined) {
+        bot.log.error(
+          `Could not find referendumId in block ${tipResult.blockHash}. EncodedProposal="${encodedProposal}". Polkassembly post will NOT be updated.`,
+        );
+        return tipResult;
+      }
+      referendumId = maybeReferendumId;
+    } catch (e) {
+      bot.log.error(
+        `Error when trying to find referendumId in block ${tipResult.blockHash}. EncodedProposal="${encodedProposal}". Polkassembly post will NOT be updated.`,
+      );
+      bot.log.error(e.message);
+      return tipResult;
+    }
```

Comment:
Should we maybe move this to a different method? This and the following line makes it a bit harder to read and combobulates the file.

```suggestion
const getReferendumId = (tipResult: number): Promise<number | null> => {
    try {
      const maybeReferendumId = await getReferendumId(await api.at(tipResult.blockHash), encodedProposal);
      if (maybeReferendumId === undefined) {
        bot.log.error(
          `Could not find referendumId in block ${tipResult.blockHash}. EncodedProposal="${encodedProposal}". Polkassembly post will NOT be updated.`,
        );
        return null;
      }
      return maybeReferendumId;
    } catch (e) {
      bot.log.error(
        `Error when trying to find referendumId in block ${tipResult.blockHash}. EncodedProposal="${encodedProposal}". Polkassembly post will NOT be updated.`,
      );
      bot.log.error(e.message);
      return null;
    }
}

let referendumId: number = getReferendumId(tipResult) ?? tipResult;
```